apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Transaction.java

```diff
@@ -79,11 +79,6 @@
     @Nullable
     private String result;
 
-    /**
-     * Noop transactions won't be reported at all, in contrast to non-sampled transactions.
-     */
-    private boolean noop;
```

Comment:
It will be great to get rid of this state, just make sure that the combination of `discardRequested`, `discardable`  and `sampled` provide the same meaning as `noop`.

Reply:
It wasn't equivalent. I've reverted the removal of the noop flag. It's also something only needed for the OT API... So maybe remove in a dedicated PR but this one changes too much already.